src/frontend/Semantic_check.ml

```diff
@@ -374,11 +374,21 @@ let semantic_check_binop loc op (le, re) =
 
 let to_exn v =
   v |> Validate.to_result
-  |> Result.map_error ~f:Fmt.(to_to_string @@ list ~sep:cut Semantic_error.pp)
+  |> Result.map ~f:(fun (result, warnings) ->
+         ( result
+         , Fmt.(to_to_string @@ list ~sep:cut Semantic_warning.pp) warnings )
+     )
+  |> Result.map_error
+       ~f:
+         Fmt.(
+           to_to_string
+           @@ pair ~sep:cut
+                (list ~sep:cut Semantic_error.pp)
+                (list ~sep:cut Semantic_warning.pp))
```

Comment:
I think maybe the `|> fst` should go inside this `to_exn` function rather than at its two use sites?